processing/src/main/java/io/druid/query/timeseries/TimeseriesQueryQueryToolChest.java

```diff
@@ -131,72 +136,96 @@ public TimeseriesQueryQueryToolChest(
       }
     };
 
-    return new QueryRunner<Result<TimeseriesResultValue>>()
-    {
-      @Override
-      public Sequence<Result<TimeseriesResultValue>> run(
-          final QueryPlus<Result<TimeseriesResultValue>> queryPlus,
-          final Map<String, Object> responseContext
-      )
-      {
-        final TimeseriesQuery query = (TimeseriesQuery) queryPlus.getQuery();
-        final Sequence<Result<TimeseriesResultValue>> baseResults = resultMergeQueryRunner.run(
-            queryPlus.withQuery(
-                queryPlus.getQuery()
-                         .withOverriddenContext(
-                             ImmutableMap.of(TimeseriesQuery.CTX_GRAND_TOTAL, false)
-                         )
-            ),
-            responseContext
-        );
+    return (queryPlus, responseContext) -> {
+      final TimeseriesQuery query = (TimeseriesQuery) queryPlus.getQuery();
+      final Sequence<Result<TimeseriesResultValue>> baseResults = resultMergeQueryRunner.run(
+          queryPlus.withQuery(
+              queryPlus.getQuery()
+                       .withOverriddenContext(
+                           ImmutableMap.of(TimeseriesQuery.CTX_GRAND_TOTAL, false)
+                       )
+          ),
+          responseContext
+      );
+
+      final Sequence<Result<TimeseriesResultValue>> finalSequence;
+
+      if (query.getGranularity().equals(Granularities.ALL) && !query.isSkipEmptyBuckets()) {
+        //Usally it is NOT Okay to materialize results via toList(), but Granularity is ALL thus we have only one record
+        finalSequence = baseResults.toList().isEmpty() ? Sequences.simple(Collections.singletonList(
```

Comment:
I believe this will end up running the query on the cluster twice (once for the `toList` and then another time, later, when the `finalSequence` is accumulated). We should avoid this by caching `baseResults.toList()` and doing a `Sequences.simple` out of it once it's computed.

Reply:
sure make sense